tests/kubernetes_test.py

```diff
@@ -73,3 +79,74 @@ def test_handle_event_exit_early_on_misrouted_event(mock_kubernetes_task):
     # we log before actually doing anything with an event, so this not being called means
     # we exited early
     assert not mock_log_event_info.called
+
+
+def test_create_task_disabled():
+    cluster = KubernetesCluster("kube-cluster-a:1234", enabled=False)
+    mock_serializer = mock.MagicMock()
+
+    task = cluster.create_task(action_run_id="action_a", serializer=mock_serializer,)
+
+    assert task is None
+
+
+def test_create_task(mock_kubernetes_cluster):
+    mock_serializer = mock.MagicMock()
+
+    task = mock_kubernetes_cluster.create_task(action_run_id="action_a", serializer=mock_serializer,)
+
+    assert task is not None
+
+
+def test_create_task_with_task_id(mock_kubernetes_cluster):
+    mock_serializer = mock.MagicMock()
+
+    task = mock_kubernetes_cluster.create_task(action_run_id="action_a", serializer=mock_serializer, task_id="yay.1234")
+
+    assert task.get_config().pod_name == "yay.1234"
+
+
+def test_create_task_with_invalid_task_id(mock_kubernetes_cluster):
+    mock_serializer = mock.MagicMock()
+
+    task = mock_kubernetes_cluster.create_task(action_run_id="action_a", serializer=mock_serializer, task_id="boo")
+
+    assert task is None
+
+
+def test_process_event_task(mock_kubernetes_cluster):
+    event = mock_event_factory(task_id="abc.123", platform_type="mock_type")
+    mock_kubernetes_task = mock.MagicMock(spec_set=KubernetesTask)
+    mock_kubernetes_task.get_kubernetes_id.return_value = "abc.123"
+    mock_kubernetes_cluster.tasks["abc.123"] = mock_kubernetes_task
+
+    mock_kubernetes_cluster.process_event(event)
+
+    mock_kubernetes_task.handle_event.assert_called_once_with(event)
+
+
+def test_process_event_task_invalid_id(mock_kubernetes_cluster):
+    event = mock_event_factory(task_id="hwat.dis", platform_type="mock_type")
+    mock_kubernetes_task = mock.MagicMock(spec_set=KubernetesTask)
+    mock_kubernetes_task.get_kubernetes_id.return_value = "abc.123"
+    mock_kubernetes_cluster.tasks["abc.123"] = mock_kubernetes_task
+
+    mock_kubernetes_cluster.process_event(event)
+
+    assert mock_kubernetes_task.handle_event.call_count == 0
+
+
+def test_stop_default(mock_kubernetes_cluster):
+    # When stopping, tasks should not exit. They will be recovered
+    mock_task = mock.MagicMock()
+    mock_kubernetes_cluster.tasks = {"task_id": mock_task}
+    mock_kubernetes_cluster.stop()
+    assert mock_kubernetes_cluster.deferred is None
+    assert mock_task.exited.call_count == 0
+    assert len(mock_kubernetes_cluster.tasks) == 1
+
```

Comment:
Can we also test set_enabled? Then we can check if connect() and stop() are being called properly.

Reply:
done!